test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs

```diff
@@ -475,6 +475,211 @@ private static void CreateXmlFile(string configFilePath, string nugetConfigStrin
             }
         }
 
+        [PlatformTheory(Platform.Windows)]
+        [InlineData(@"https://source.test", true)]
+        [InlineData(@"\\myserver\packages", false)]
+        public void Sources_WhenAddingSourceWithProtocolVersion_GotAdded(string source, bool shouldWriteProtocolVersion)
+        {
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                var workingPath = pathContext.WorkingDirectory;
+                var settings = pathContext.Settings;
+
+                // Arrange
+                var args = new string[]
+                {
+                    "nuget",
+                    "add",
+                    "source",
+                    source,
+                    "--name",
+                    "test_source",
+                    "--configfile",
+                    settings.ConfigPath,
+                    "--protocol-version",
+                    "3"
+                };
+
+                // Act
+                var result = _fixture.RunDotnetExpectSuccess(workingPath, string.Join(" ", args));
+
+                // Assert
+                Assert.True(result.ExitCode == 0);
+                var loadedSettings = Settings.LoadDefaultSettings(root: workingPath, configFileName: null, machineWideSettings: null);
+                var packageSourcesSection = loadedSettings.GetSection("packageSources");
+                var sourceItem = packageSourcesSection?.GetFirstItemWithAttribute<SourceItem>("key", "test_source");
+                var expectedProtocolVersion = shouldWriteProtocolVersion ? "3" : null;
+                Assert.Equal(expectedProtocolVersion, sourceItem.ProtocolVersion);
+            }
+        }
+
+        [PlatformTheory(Platform.Windows)]
+        [InlineData("1", false)]
+        [InlineData("2", true)]
+        [InlineData("3", true)]
+        [InlineData("4", false)]
+        [InlineData("5", false)]
+        public void Sources_WhenAddingSourceWithProtocolVersion_ProtocolVersionIsValidated(string protocolVersion, bool shouldSucceed)
```

Comment:
Why is this test marked as Windows only? I don't see anything obvious about this test that should prevent it from working on Linux or Mac.